tools/testsys/src/vmware_k8s.rs

```diff
@@ -0,0 +1,241 @@
+    async fn starting_image_id(&self, crd_input: &CrdInput) -> Result<String> {
+        Ok(format!(
+            "bottlerocket-{}-{}-{}",
+            crd_input.variant,
+            crd_input.arch,
+            crd_input
+                .starting_version
+                .as_ref()
+                .context(error::InvalidSnafu {
+                    what: "The starting version must be provided for migration testing"
+                })?
+        ))
+    }
+
+    /// Create an EKS cluster CRD with the `cluster_name` in `cluster_input`.
+    async fn cluster_crd<'a>(&self, cluster_input: ClusterInput<'a>) -> Result<CreateCrdOutput> {
+        let control_plane_endpoint = cluster_input
+            .crd_input
+            .config
+            .control_plane_endpoint
+            .as_ref()
+            .context(error::InvalidSnafu {
+                what: "The control plane endpoint is required for vmware cluster creation.",
+            })?;
+        let labels = cluster_input.crd_input.labels(btreemap! {
+            "testsys/type".to_string() => "cluster".to_string(),
+            "testsys/cluster".to_string() => cluster_input.cluster_name.to_string(),
+            "testsys/controlPlaneEndpoint".to_string() => control_plane_endpoint.to_string(),
+        });
+
+        // Check if the cluster already has a crd
+        if let Some(cluster_crd) = cluster_input
+            .crd_input
+            .existing_crds(
+                &labels,
+                &[
+                    "testsys/cluster",
+                    "testsys/type",
+                    "testsys/controlPlaneEndpoint",
+                ],
+            )
+            .await?
+            .pop()
+        {
+            return Ok(CreateCrdOutput::ExistingCrd(cluster_crd));
+        }
+
+        // Check if an existing cluster is using this endpoint
+        let existing_clusters = cluster_input
+            .crd_input
+            .existing_crds(&labels, &["testsys/type", "testsys/controlPlaneEndpoint"])
+            .await?;
+
+        let cluster_version =
+            K8sVersion::from_str(cluster_input.crd_input.variant.version().context(
+                error::MissingSnafu {
+                    item: "K8s version".to_string(),
+                    what: "aws-k8s variant".to_string(),
+                },
+            )?)
+            .map_err(|_| error::Error::K8sVersion {
+                version: cluster_input.crd_input.variant.to_string(),
+            })?;
+
+        let vsphere_k8s_crd = VSphereK8sClusterConfig::builder()
+            .name(cluster_input.cluster_name)
+            .control_plane_endpoint_ip(control_plane_endpoint)
+            .creation_policy(CreationPolicy::IfNotExists)
+            .version(cluster_version)
+            .ova_name(self.image_id(cluster_input.crd_input)?)
+            .tuf_repo(
+                cluster_input
+                    .crd_input
+                    .tuf_metadata()
```

Comment:
I should have caught this earlier but `tuf_metadata` is slightly confusing since it's suppose to return both the TUF metadata base URL as well as TUF repository targets base URL. Maybe `tuf_info` instead?

Reply:
I think that belongs in a different pr though right?

Reply:
Sure that's fine, but can we track this as a quick follow up?